x-pack/plugins/fleet/server/services/agents/update_agent_tags.ts

```diff
@@ -87,12 +87,29 @@ async function updateTagsBatch(
 ): Promise<{ items: BulkActionResult[] }> {
   const errors: Record<Agent['id'], Error> = { ...outgoingErrors };
 
+  const getNewTags = (agent: Agent): string[] => {
+    const existingTags = agent.tags ?? [];
+
+    if (tagsToAdd.length === 1 && tagsToRemove.length === 1) {
```

Comment:
optimizing logic for the rename use case (one tag to add, one to remove), so that the renamed tag is not jumping to a different place on UI.
<img width="435" alt="image" src="https://user-images.githubusercontent.com/90178898/177946477-6d89b63a-02f5-485c-94d7-64ebed48d3da.png">
<img width="476" alt="image" src="https://user-images.githubusercontent.com/90178898/177946538-21d9b1c2-4e74-454a-bf02-9a99df36f8a3.png">

Reply:
Nice thank you for clarifying the solution here